### ISSUE TITLE
NullPointerException when generating Schema with @JsonTypeInfo annotations

### ISSUE THREAD

**Issue by maciejkujawski** [springdoc/springdoc-openapi#102]
Hi,

I got error during schema generation when one of schema Classes are annotated with: `@JsonTypeInfo`

Example:

```
@JsonTypeInfo(use = Id.NAME, property = "type")
@JsonSubTypes({
    @JsonSubTypes.Type(value = A.class, name = "A"),
    @JsonSubTypes.Type(value = B.class, name = "B"),
    @JsonSubTypes.Type(value = C.class, name = "C")
})
@Schema(name = "ABC", subTypes = {A.class, B.class, C.class})
public abstract class ABC {
```

Null pointer is at:

```
Caused by: java.lang.NullPointerException: null
	at io.swagger.v3.core.jackson.ModelResolver.resolveDiscriminatorProperty(ModelResolver.java:1643)
```

The second condition is checking model instead of modelToUpdate and then: `model.getRequired()` is null?

```
if (modelToUpdate.getRequired() == null || !model.getRequired().contains(typeInfoProp)) {
```

**Comment by bnasslahsen** [comment]
Hi, 

 This is not an issue on springdoc-openapi.
 Please raise your issue on swagger-core, github issue:
- https://github.com/swagger-api/swagger-core/issues

**Comment by venaud** [comment]
Hello,

swagger-core fixed this issue in release 2.0.10.

Could you bump its version from 2.0.9 to 2.0.10?

For gradle users, the temporary fix is:
```
implementation('org.springdoc:springdoc-openapi-ui:1.1.48')
    implementation('io.swagger.core.v3:swagger-core:2.0.10') {
        force = true
    }
```
